tools/scylla-sstable-crawler.cc

```diff
@@ -0,0 +1,693 @@
+            ssc_log.info("Validated {}: {}", sst ? sst->get_filename() : "the stream", valid ? "valid" : "invalid");
+            return stop_iteration::no;
+        });
+    }
+};
+
+template <typename SstableConsumer>
+class sstable_consumer_operation : public operation {
+    virtual std::string_view name() const override { return SstableConsumer::name; }
+    virtual std::string_view description() const override { return SstableConsumer::description; }
+    virtual void operator()(schema_ptr schema, reader_permit permit, const std::vector<sstables::shared_sstable>& sstables,
+            const partition_set& partitions, const options& opts) const override {
+        auto consumer = std::make_unique<SstableConsumer>(schema, permit);
+        consumer->on_start_of_stream().get();
+        consume_sstables(schema, permit, sstables, opts.merge, [&, &consumer = *consumer] (flat_mutation_reader& rd, sstables::sstable* sst) {
+            return consume_reader(std::move(rd), consumer, sst, partitions, opts.no_skips);
+        });
+        consumer->on_end_of_stream().get();
+    }
+};
+
+// Not really shared but we need copyable types for initializer_list<>
+const std::vector<shared_ptr<operation>> operations{
+    make_shared<sstable_consumer_operation<dumping_consumer>>(),
+    make_shared<sstable_consumer_operation<writetime_histogram_collecting_consumer>>(),
+    make_shared<sstable_consumer_operation<custom_consumer>>(),
+    make_shared<validate_operation>(),
+};
+
+} // anonymous namespace
+
+int main(int argc, char** argv) {
+    app_template::config app_cfg;
+    app_cfg.name = app_name;
+
+    const auto description_template =
+R"(scylla-sstable-crawler - a multifunctional command-line tool to examine the content of sstables.
+
+Allows examining the contents of sstables with various built-in tools
+(operations). The sstables to-be-examined are passed as positional
+command line arguments. Sstables will be processed by the selected
+operation one-by-one (can be changed with --merge). Any number of
+sstables can be passed but mind the open file limits. Pass the full path
+to the data component of the sstables (*-Data.db). For now it is
+required that the sstable is found at a valid data path:
+
+    /path/to/datadir/{{keyspace_name}}/{{table_name}}-{{table_id}}/
+
+The schema to read the sstables is read from a schema.cql file. This
+should contain the keyspace and table definitions, as well as any UDTs
+used.
+Filtering the sstable()s to process only certain partition(s) is
+supported via the --partition and --partitions-file command line flags.
+Partition keys are expected to be in the hexdump format used by scylla
+(hex representation of the raw buffer).
+Operations write their output to stdout, or file(s). The tool logs to
+stderr, with a logger called {}.
+
+The supported operations are:
+{}
+
+Examples:
+
+# dump the content of the sstable
+$ scylla-sstable-crawler dump /path/to/md-123456-big-Data.db
```

Comment:
Can `dump` be passed as a free option? Isn't the syntax `--operation=dump`

Reply:
Also applies to the other examples below...

Reply:
Hmm, it looks like you changed the syntax.

Reply:
I changed it to be a positional argument. It is somewhat less typing. There are three alternatives:
* `--operation=${operation}`
* `--${operation}` (used by `scylla-types`)
* `$operation` (positional argument)
I'm somewhat undecided. The syntax I used prior to publishing requires some extra (unnecessary) typing. Maybe the second option is better? It is also used by an existing tool already.

Reply:
The 2nd option looks best to me too.

Reply:
Also I found (as noted below) that boost handles multiple positional arguments rather poorly, or maybe there just isn't a good way to handle them. Without the associating flag, it is really hard to decide which positional option is which, especially when you have one with variable amount of instances. I think I'll go for option 2 from the list above, it is less surprises overall.

Reply:
Ok, 2nd option it is then.